functional.lua

```diff
@@ -263,4 +274,16 @@ function functional.find_match(t, f)
 	return nil
 end
 
+-- returns a table where the elements in t are grouped by the result of f on each element.
+function functional.group_by(t, f)
+	local result = {}
+	for i, v in ipairs(t) do
+		local group = f(v)
+		if result[group] == nil then
+			result[group] = {}
+		end
+		table.insert(result[group], v)
+	end
```

Comment:
Probably needs to `return result` here 😉

Reply:
Curse you Lisp for having me forget how to write `return` 😞